amethyst_renderer/src/sprite.rs

```diff
@@ -24,19 +27,38 @@ pub struct SpriteSheet {
 }
 
 impl Asset for SpriteSheet {
-    const NAME: &'static str = "renderer::Sprite";
+    const NAME: &'static str = "renderer::SpriteSheet";
     type Data = Self;
     type HandleStorage = VecStorage<Handle<Self>>;
 }
 
-/// A description of a frame in a sprite sheet.
+impl From<SpriteSheet> for AssetsResult<ProcessingState<SpriteSheet>> {
+    fn from(sprite_sheet: SpriteSheet) -> AssetsResult<ProcessingState<SpriteSheet>> {
+        Ok(ProcessingState::Loaded(sprite_sheet))
+    }
+}
+
+/// Dimensions and texture coordinates of each sprite in a sprite sheet.
+#[derive(Clone, Debug, PartialEq, Serialize, Deserialize)]
+pub struct Sprite {
+    /// Pixel width of the sprite
+    pub width: f32,
+    /// Pixel height of the sprite
+    pub height: f32,
```

Comment:
Isn't this already defined by texture coordinates?

Also, I didn't check too much of the implementation (yet), but I personally use a [0,1,1,0] camera for my games. I expect to be able to create a (0,1,1,0) mesh and scale it down using transform after that. Is this something possible using this?

Reply:
I feel like I might be unclear. What I mean to ask is: Is this actual pixel size, or is it relative to the camera's dimensions matrix?

Reply:
> Isn't this already defined by texture coordinates?

I don't think it is -- you can have a big texture with 1 sprite, or a small texture with 1 sprite, and the texture coordinates will be largely different since they are proportions of the texture. You'd still need the sprite dimensions to know how much space it occupies in the world.

> [0,1,1,0] camera

what do the numbers correspond to? (forgive my lapse in knowledge :confused:). It's the actual pixel size on the texture. I don't think I understand/remember the camera dimension's matrix without re-reading =o

Reply:
left right top bottom

I want to know if the sprite pass takes the camera's matrix into account when rendering, or it uses the raw pixel size to pixels on the screen.

Reply:
ohh, yeap it does take the camera's matrix into account. So if the camera resolution is halved then you get double sized sprites.

Oh ya, should the `TextureCoordinates` use left right top bottom as the field order then? I swapped it when inverting the Y axis direction

Reply:
If you got inspiration from SpriteRenderData, then yes, the Y axis is wrong in it.

Reply:
Sweet, also I'll keep the parameter order to `(left, right, bottom, top)` because that's how `glOrtho` does it<sup>[[1]](https://www.khronos.org/registry/OpenGL-Refpages/gl2.1/xhtml/glOrtho.xml)[[2]](https://docs.microsoft.com/en-us/windows/desktop/opengl/glortho)</sup>

Reply:
We should normalize it in the engine. Its often unclear which order we choose.